dom.bs

```diff
@@ -1156,15 +1167,26 @@ for discussion).
   <p class="note"><var>legacy target override flag</var> is only used by HTML and only when
   <var>target</var> is a {{Window}} object.
 
- <li>Let <var>relatedTarget</var> be the result of <a>retargeting</a> <var>event</var>'s
- <a for=Event>relatedTarget</a> against <var>target</var> if <var>event</var>'s
- <a for=Event>relatedTarget</a> is non-null, and null otherwise.
+ <li><p>Let <var>clearTargetsPostDispatch</var> be true if <var>target</var>, <var>event</var>'s
+ <a for=Event>relatedTarget</a>, or an {{EventTarget}} object in <var>event</var>'s
+ <a for=Event>touch target list</a> is a <a>node</a> and its <a for=tree>root</a> is a
+ <a for=/>shadow root</a>, and false otherwise.
+
+ <li><p>Let <var>relatedTarget</var> be the result of <a>retargeting</a> <var>event</var>'s
+ <a for=Event>relatedTarget</a> against <var>target</var>.
 
  <li><p>If <var>target</var> is <var>relatedTarget</var> and <var>target</var> is not
  <var>event</var>'s <a for=Event>relatedTarget</a>, then return true.
+ <!-- XXX Since we return early, we might want to unset more flags here. See #402. -->
```

Comment:
Should we also unset target et al here if clearTargetsPostDispatch is true? It seems that would be consistent... (If that's the case it seems we should turn this into a short algorithm that we invoke here and later on.)

Reply:
I have never thought about this case seriously so far. In this early return, I assume that any event listener is never called.

In this case, we don't have to worry about any leaking of a node in a shadow tree via an event object. Users don't have any change to access an event object, other than the original event dispatcher.

btw, the consistency may matter. I'm fine to reset target, relatedTarget, and touch target list here too.

Reply:
Yeah, I'm mostly wondering about it from a consistency perspective. I could see you'd have some custom dispatching code and later store the event object after dispatch in a global place and maybe you don't want things to be leaked there ever. Doesn't seem like a sound setup, but other than this case we do provide the guarantee that nothing would leak.

Reply:
I've now made this change.